### DISCUSSION TITLE
Installing in Proxmox NAT

### DISCUSSION THREAD

**Discussion by mkikets99** [postalserver/postal#2712] [Help with using Postal]
I am going through instruction and now I am at this point:

> Run the command below and replace postal.yourdomain.com with the actual hostname you want to access your Postal web interface at. Make sure you have set up this domain with your DNS provider before continuing.
> 
> ```
> postal bootstrap postal.yourdomain.com
> ```

I am curious:
Can I provide local DNS (configured on HaProxy/pfSence) as web config DNS (something like `postal.example.lan`) and than configure additional global domains (`mail.example.com`)?

**Comment by willpower232** [comment]
Postal needs one domain to identify itself and then you can add sending/receiving domains to Postal which reference the initial domain.

For example, you could go for `postal.yourcompany.com` and then add `yourclient.com` in the UI which would have SPF and/or MX DNS records referring to `postal.yourcompany.com`.

**Reply by mkikets99** [reply]
You meaning that `First domain` mustbe globally accessible, right?

**Reply by willpower232** [reply]
Postal expects that, yes.

Fundamentally you can set up any combination of domains you want and ignore Postal validating the domains no problem but it gives you a good clue where problems might be if you go along with it.

You could of course keep emailing [mail-tester.com](https://www.mail-tester.com/) for debugging your DNS but that would get a bit boring after a while.

**Reply by pdifeo** [reply]
When there are services behind NAT, it is always better to have an external DNS and one inside the NAT. The configuration work is double, but you avoid problems with public and private IPs.

Personally, I have installed Postal Server on a Proxmox host, but the internal dns is managed by the firewall that manages the NAT.

**Reply by Pacerino** [reply]
@pdifeo small question here because i read that you've installed Postal on a Proxmox host. Do you use the IP Pools and if so, how did you managed it? I'm assuming you're using some kind of Router/Bridged Setup with pfSense/OPNSense?

**Reply by pdifeodfware** [reply]
I did not quite understand your question. For Postal I use only one public IP address in NAT on internal IP. Yes, I use PFSense. With PFSense you can use multiple IP addresses on the WAN side by creating Virtual IPs.
